services/CurrentTime/source/CurrentTimeService.cpp

```diff
@@ -68,12 +104,19 @@ void CurrentTimeService::onDataRead(GattReadAuthCallbackParams *read_request) {
     read_request->authorizationReply = AUTH_CALLBACK_REPLY_SUCCESS;
 }
 
-void CurrentTimeService::onDataWritten(GattWriteAuthCallbackParams *write_request) {
+void CurrentTimeService::onDataWritten(GattWriteAuthCallbackParams *write_request)
+{
     const uint8_t *data = write_request->data;
 
     struct tm remote_time_tm{};
 
-    deserialize(&remote_time_tm, data);
+    write_request->authorizationReply = AUTH_CALLBACK_REPLY_SUCCESS;
+
+    if (!deserialize(&remote_time_tm, data)) {
+        /* The alert level is out of range
+         * */
+        write_request->authorizationReply = static_cast<GattAuthCallbackReply_t>(0xFF);
```

Comment:
you send an error but the code continues past the if statement

how about:
```
if (deserialise) {
  call handler
} else {
  send error
}
```

Reply:
also you don't need ```static_cast<GattAuthCallbackReply_t>(0xFF)``` any more, Vincent added the missing AUTH_CALLBACK_REPLY_ATTERR_OUT_OF_RANGE error and others and they've been merged to mbedos